arangod/Cluster/ClusterInfo.cpp

```diff
@@ -2878,7 +2879,41 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
 
   {
     while (!application_features::ApplicationServer::isStopping()) {
-      int tmpRes = dbServerResult->load(std::memory_order_acquire);
+      int tmpRes = dbServerResult.load(std::memory_order_acquire);
+
+      if (tmpRes < 0) {
+        // index has not shown up in Current yet,  follow up check to
+        // ensure it is still in plan (not dropped between iterations)
+
+        AgencyCommResult result = _agency.sendTransactionWithFailover(
+            AgencyReadTransaction(AgencyCommManager::path(planIndexesKey)));
+
+        if (result.successful()) {
+          auto indexes = result.slice()[0].get(
+              std::vector<std::string>{AgencyCommManager::path(), "Plan",
+                                       "Collections", databaseName,
+                                       collectionID, "indexes"});
+
+          bool found = false;
+          if (indexes.isArray()) {
+            for (size_t i = 0; i < indexes.length(); i++) {
+              VPackSlice v = indexes.at(i);
```

Comment:
VPackArrayIterator

Reply:
For some reason using the `VPackArrayIterator` and a `continue` to skip was terminating the loop early. This way worked reliably. Does it make much difference functionally?

Reply:
array iterator is faster ?

Reply:
Okay, let me try again and see if it works reliably now. (Might have been something else confounding my investigation.)